### PR TITLE
implement forgot password functionality

### PR DESCRIPTION
**Pull request by yulia-dnistrian**
Description:

Linting:

- [x] No linting errors

Tests:

- [x] E2E tests (npm test run with `e2e`)
- [x] Manual tests

Documentation:

- [ ] Requires documentation updates

### CODE REVIEW

File: src/containers/ChangePasswordPage.less [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this license. If not, see
+ *
+ *     https://www.gnu.org/licenses/
+ *
+ *
+ */
+
+@import '../style/form-component-style.less';
```
Comment (shaunmaharaj) [inline src/containers/ChangePasswordPage.less]: This fails to build. Did you forget to push this file?

File: src/containers/ResetPasswordPage.less [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this license. If not, see
+ *
+ *     https://www.gnu.org/licenses/
+ *
+ *
+ */
+
+@import '../style/form-component-style.less';
```
Comment (shaunmaharaj) [inline src/containers/ResetPasswordPage.less]: This fails to build. Did you forget to push this file?

File: src/components/appmodallogin.main.jsx [hunk 3]
```diff
@@ -162,6 +169,7 @@ class AppModalLoginMain extends React.Component {
                   {
                     (isLoading) ? <div className="miniLoader" /> : ('')
                   }
+                  <button type="button" className="label-link" onClick={this.resetPassword}>Forgot password?</button>
```
Comment (shaunmaharaj) [inline src/components/appmodallogin.main.jsx]: This should only show up if the `forgot password` service is available. Please add a condition to surround this button here so it doesn't show up unless the API is available.

File: src/containers/ChangePasswordPage.jsx [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    event.preventDefault();
+    const {
+      oldPassword, newPassword, newPasswordConfirmed, email,
+    } = this.state;
+    if (!oldPassword || !newPassword || !newPasswordConfirmed) {
+      this.setState({ failedSubmit: true });
+    }
+    login()
+      .then((response) => {
+        console.log(response);
```
Comment (shaunmaharaj) [inline src/containers/ChangePasswordPage.jsx]: console statement left over

File: src/containers/ChangePasswordPage.jsx [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+            Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+          },
+        })
+        .then(res => res.json())
+        .then((res) => {
+          const { email } = this.state;
+          if (res && res._defaultprofile) {
+            this.setState({ email: res._defaultprofile[0]._emails[0]._element[0].email });
+          }
+          console.log(res, email);
```
Comment (shaunmaharaj) [inline src/containers/ChangePasswordPage.jsx]: console statement left over

File: src/containers/ChangePasswordPage.jsx [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+    const {
+      oldPassword, newPassword, newPasswordConfirmed, email,
+    } = this.state;
+    if (!oldPassword || !newPassword || !newPasswordConfirmed) {
+      this.setState({ failedSubmit: true });
+    }
+    login()
+      .then((response) => {
+        console.log(response);
+        cortexFetch(`/passwords/${Config.cortexApi.scope}/reset/form/?zoom=resetpasswordaction`, {
```
Comment (shaunmaharaj) [inline src/containers/ChangePasswordPage.jsx]: Can we get this URL from the API rather than constructing it ourselves?

Comment (yulia-dnistrian) [reply src/containers/ChangePasswordPage.jsx]: Resolution of this is here https://github.com/elasticpath/react-pwa-reference-storefront/pull/267/commits/294b11874056bd3569c6ce239b5c27de1252f6d0

File: src/containers/ChangePasswordPage.jsx [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+      cortexFetch(`/?zoom=${zoomArray.join()}`,
+        {
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+          },
+        })
+        .then(res => res.json())
+        .then((res) => {
+          if (res && res._passwordresetform) {
```
Comment (shaunmaharaj) [inline src/containers/ChangePasswordPage.jsx]: This piece should be how we can validate if the forgot password feature is even available. We'll have to add a call to the appmodallogin component to make a call like this on componentdidmount and display the "forgot password" button if `passwordresetform` is available.

File: src/containers/ResetPasswordPage.jsx [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+        </div>
+        <form className="form-horizontal" onSubmit={this.submitResetEmail}>
+          <div className="form-group">
+            <div className="control-label" />
+            <div className="form-input">
+              {/* eslint-disable-next-line max-len */}
+              <p>
+                {intl.get('reset-email-txt')}
+              </p>
+              <input id="email_reset" name="EmailReset" className="form-control" type="email" value={emailReset} onChange={this.setOldPassword} />
```
Comment (shaunmaharaj) [inline src/containers/ResetPasswordPage.jsx]: Where is `setOldPassword `? Without it, this form is not editable. The value is always set to an empty string.

File: src/containers/ResetPasswordPage.jsx [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    };
+    this.setEmailReset = this.setEmailReset.bind(this);
+    this.submitResetEmail = this.submitResetEmail.bind(this);
+  }
+
+  setEmailReset(event) {
+    this.setState({ emailReset: event.target.value });
+  }
+
+  submitResetEmail(event) {
```
Comment (shaunmaharaj) [inline src/containers/ResetPasswordPage.jsx]: Where does the request for the password reset token happen?

**Pull request merged**